src/app/core/components/brim-editor.tsx

```diff
@@ -0,0 +1,120 @@
+import {Extension, EditorState, StateEffect} from "@codemirror/state"
+import {history, historyKeymap} from "@codemirror/history"
+import {foldGutter, foldKeymap} from "@codemirror/fold"
+import {lineNumbers, highlightActiveLineGutter} from "@codemirror/gutter"
+import {bracketMatching} from "@codemirror/matchbrackets"
+import {closeBrackets, closeBracketsKeymap} from "@codemirror/closebrackets"
+import {highlightSelectionMatches} from "@codemirror/search"
+import {autocompletion, completionKeymap} from "@codemirror/autocomplete"
+import {rectangularSelection} from "@codemirror/rectangular-selection"
+import {defaultHighlightStyle} from "@codemirror/highlight"
+
+const EditorWrap = styled.div`
+  width: 100%;
+  margin: 8px 0;
+  max-height: 58px;
+  overflow: scroll;
+`
+
+const editorTheme = EditorView.theme(
+  {
+    ".cm-scroller": {
+      overflow: "hidden"
+    }
+  },
+  {dark: false}
+)
+
+const baseEditorSetup: Extension = [
+  lineNumbers(),
+  highlightActiveLineGutter(),
+  highlightSpecialChars(),
+  history(),
+  foldGutter(),
+  drawSelection(),
+  dropCursor(),
+  EditorState.allowMultipleSelections.of(true),
+  defaultHighlightStyle.fallback,
+  bracketMatching(),
+  closeBrackets(),
+  autocompletion(),
+  rectangularSelection(),
+  highlightActiveLine(),
+  highlightSelectionMatches(),
+  keymap.of([
+    ...closeBracketsKeymap,
+    ...historyKeymap,
+    ...completionKeymap,
+    ...foldKeymap
+  ])
+]
+
+const BrimEditor = () => {
+  const ref = useRef<HTMLDivElement>()
+  const [view, setView] = useState<EditorView>(null)
+  const inputValue = useSelector(SearchBar.getSearchBarInputValue)
+  const [currentEditorValue, setCurrentEditorValue] = useState("")
+  const isMultiLineMode = hasNewLine(inputValue)
+  const dispatch = useDispatch()
+
+  const onKeyDown = (e) => {
+    if (e.key === "Enter" && e.metaKey) dispatch(submitSearch())
+  }
+
+  const onChangeUpdater = EditorView.updateListener.of((viewUpdate) => {
+    if (!viewUpdate.docChanged) return
+    const viewContents = viewUpdate.state.doc.toString()
+    setCurrentEditorValue(viewContents)
+    dispatch(SearchBar.changeSearchBarInput(viewContents))
+  })
+  const extensions = [editorTheme, onChangeUpdater]
+  if (isMultiLineMode) extensions.push(baseEditorSetup)
+
+  useEffect(() => {
+    if (!view) return
+    if (inputValue === currentEditorValue) return
+    view.dispatch(
+      view.state.update({
+        changes: {from: 0, to: view.state.doc.length, insert: inputValue}
+      })
+    )
+  }, [inputValue])
+  useEffect(() => {
+    if (!view) return
+    view.dispatch({
+      effects: StateEffect.reconfigure.of(extensions)
+    })
+  }, [isMultiLineMode])
+  useEffect(() => {
+    if (!ref.current) return
+    const state = EditorState.create({
+      extensions,
+      doc: inputValue
+    })
+    const newView = new EditorView({
+      state,
+      parent: ref.current
+    })
+    setView(newView)
+    newView.focus()
+  }, [ref])
```

Comment:
```suggestion
  }, [])
```

This will just run on mount because that ref object never changes its identity while the component is mounted.

You could do `[ref.current]` which will change, but the effect will run when something else causes it to render and ref.current has changed. As I'm sure you know, changing ref.current won't cause a re-render.